x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/fielddata/LabelPositionVisitor.java

```diff
@@ -7,54 +7,58 @@
 
 package org.elasticsearch.xpack.spatial.index.fielddata;
 
-import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.xcontent.ToXContentFragment;
+
+import java.util.function.BiFunction;
 
 /**
- * Get the first node of the tree and provide a point in that gemetry (point, line or triangle)
+ * Get the first node of the tree and provide a point in that geometry (point, line or triangle)
  * as a suggested label position likely to be somewhere in the middle of the entire geometry.
  *
  * TODO: We could instead choose the point closer to the centroid which improves unbalanced trees
  */
-public class LabelPositionVisitor implements TriangleTreeReader.Visitor {
+public class LabelPositionVisitor<T extends ToXContentFragment> implements TriangleTreeReader.Visitor {
```

Comment:
Do we need to extend ToXContentFragment here? I think we don't need any method from that interface.